clients/ui/frontend/src/app/pages/settings/roleBinding/RoleBindingPermissionsTableRow.tsx

```diff
@@ -0,0 +1,259 @@
+import * as React from 'react';
+import { ActionsColumn, Tbody, Td, Tr } from '@patternfly/react-table';
+import {
+  Button,
+  Popover,
+  Split,
+  SplitItem,
+  Content,
+  Timestamp,
+  TimestampTooltipVariant,
+  Tooltip,
+  Truncate,
+} from '@patternfly/react-core';
+import {
+  CheckIcon,
+  OutlinedQuestionCircleIcon,
+  TimesIcon,
+  EllipsisVIcon,
+} from '@patternfly/react-icons';
+import {
+  DashboardPopupIconButton,
+  relativeTime,
+  RoleBindingKind,
+  RoleBindingSubject,
+} from 'mod-arch-shared';
+import useUser from '~/app/hooks/useUser';
+import {
+  castRoleBindingPermissionsRoleType,
+  firstSubject,
+  roleLabel,
+  isCurrentUserChanging,
+} from './utils';
+import { RoleBindingPermissionsRoleType } from './types';
+import RoleBindingPermissionsNameInput from './RoleBindingPermissionsNameInput';
+import RoleBindingPermissionsPermissionSelection from './RoleBindingPermissionsPermissionSelection';
+import RoleBindingPermissionsChangeModal from './RoleBindingPermissionsChangeModal';
+
+type RoleBindingPermissionsTableRowProps = {
+  roleBindingObject?: RoleBindingKind;
+  subjectKind: RoleBindingSubject['kind'];
+  isEditing: boolean;
+  isAdding: boolean;
+  defaultRoleBindingName?: string;
+  permissionOptions: {
+    type: RoleBindingPermissionsRoleType;
+    description: string;
+  }[];
+  typeAhead?: string[];
+  onChange: (name: string, roleType: RoleBindingPermissionsRoleType) => void;
+  onCancel: () => void;
+  onEdit?: () => void;
+  onDelete?: () => void;
+};
+
+const defaultValueName = (obj: RoleBindingKind) => firstSubject(obj);
+const defaultValueRole = (obj: RoleBindingKind) =>
+  castRoleBindingPermissionsRoleType(obj.roleRef.name);
+
+const RoleBindingPermissionsTableRow: React.FC<RoleBindingPermissionsTableRowProps> = ({
+  roleBindingObject: obj,
```

Comment:
Same thing here, good call removing the projects reference, but it would be needed soon enough, add TODOs to add it later.